### ISSUE TITLE
Checklist not loading from Template

### ISSUE THREAD

**Issue by DoktorToast** [akiko-pusu/redmine_issue_templates#141]
Hey, first I wanna thank you for your awesome plugin.
However, I'm experiencing an issue I'm not able to fix by myself:

I have Checklist Plugin (Free Version) installed, but when I'm adding Checklist Items to a Template they just disappear when I'm loading the template again and are not showing up when using the Template on an issue.

Is this because I have the Free Version only or is this a bug? Everything else works fine!

EDIT: Checking my Redmine Log, I'm able to see that the checklist seems to be saved....but they're not even showing up on the template overview directly after saving the template.
On global templates, everything works fine. Only project templates won't work.

![logscreen](https://user-images.githubusercontent.com/30146490/28162994-898fce44-67c9-11e7-9b72-9a40c37a0684.PNG)

**Comment by akiko-pusu** [comment]
@DoktorToast, Thanks for using my plugin and reporting issue.
Though I had added feature related to Checklist plugin, there was no feedback still now. So that yours is quite helpful for me.

Could you please let me know the version of Redmine itself, IssueTemplate, and Checklist.
Also I’m going to check / review its behavior.

**Comment by DoktorToast** [comment]
Wow, this was a fast response!
Environment:
  Redmine version                3.3.1.stable
  Ruby version                   2.3.3-p222 (2016-11-21) [x86_64-linux-gnu]
  Rails version                  4.2.7.1
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.9.5
  Git                            2.11.0
  Filesystem                     
Redmine plugins:
  redmine_checklists             3.1.6
  redmine_favourite_projects     0.9.1
  redmine_issue_templates        0.1.6
  redmine_my_page                0.1.10

Here's everything I can give you as information. I hope it helps!

**Comment by akiko-pusu** [comment]
Thanks for your information.
Please let me know the repository url or download site of redmine_checklists 3.1.6.
It seems there are some repositories of redmine_checklists, maybe which are forked.

**Comment by DoktorToast** [comment]
It's the free/light version from https://www.redmineup.com/
There's no repo download available I guess. There's another checklist plugin though, but this one isn't compatible with your plugin (redmine_issue_checklist).

**Comment by akiko-pusu** [comment]
I could reproduce this defect, thank again.
It seems this defect happens under project template, but global template can save checklists.
Please wait for a while?

**Comment by akiko-pusu** [comment]
@DoktorToast 
 This is a workaround for this bug. (Degrade bug on commit: f06f26b37)

Could you please edit app/controllers/issue_templates_controller.rb line: 250,
From: :enabled_sharing, :checklists)
To: :enabled_sharing, checklists: [])

Hope this will be help.

```
diff --git a/app/controllers/issue_templates_controller.rb b/app/controllers/issue_templates_controller.rb
index b8b6a39..d143b07 100644
--- a/app/controllers/issue_templates_controller.rb
+++ b/app/controllers/issue_templates_controller.rb
@@ -247,6 +247,6 @@ class IssueTemplatesController < ApplicationController
 
   def template_params
     params.require(:issue_template).permit(:tracker_id, :title, :note, :issue_title, :description, :is_default,
-                                           :enabled, :author_id, :position, :enabled_sharing, :checklists)
+                                           :enabled, :author_id, :position, :enabled_sharing, checklists: [])
   end
 end

```

**Comment by DoktorToast** [comment]
Thank you very much for this fast fix :)
I won't be able to try in the next 5 days but I'll give you feedback as soon as i tested the workaround.

Thank you!

**Comment by akiko-pusu** [comment]
This bug was fixed on commit: 302456b and now applied on master branch.
(Version number is still v0.1.6)

**Comment by DoktorToast** [comment]
It works fine now.
Thank you very much!
